Sources/Tracing/Docs.docc/Guides/TraceYourApplication.md

```diff
@@ -0,0 +1,443 @@
+            ]
+        ),
+      // ...
+    ]
+)
+
+```
+
+In order to gain a whole set of well-typed attributes which are pre-defined by the [OpenTelemetry](http://opentelemetry.io) initiatve. 
+
+For example, like these for HTTP:
+
+```swift
+attributes.http.method = "GET"
+attributes.http.url = "https://www.swift.org/download"
+attributes.http.target = "/download"
+attributes.http.host = "www.swift.org"
+attributes.http.scheme = "https"
+attributes.http.statusCode = 418
+attributes.http.flavor = "1.1"
+attributes.http.userAgent = "test"
+attributes.http.retryCount = 42
+```
+
+or these, for database operations–which can be very useful to detect slow queries in your system:
+
+```swift
+attributes.db.system = "postgresql"
+attributes.db.connectionString = "test"
+attributes.db.user = "swift"
+attributes.db.statement = "SELECT name, display_lang FROM Users WHERE id={};"
+```
+
+Using such standardized attributes allows you, and other developers of other services you interact with, have a consistent and simple to search by attribute namespace.
+
+#### Declaring your own type-safe Span Attributes
+
+You can define your own type-safe span attributes, which is useful if your team or company has a certain set of attributes you like to set in all services; This way it is easier to remember what attributes one should be setting, and what their types should be, because the attributes pop up in your favorite IDE's autocompletion.
+
+Doing so requires some boilerplate, but you only have to do this once, and later on the use-sites of those attributes look quite neat (as you've seen above). Here is how you would declare a custom `http.method` nested attribute:
+
+```swift
+extension SpanAttributes {
+    /// Semantic conventions for HTTP spans.
+    ///
+    /// OpenTelemetry Spec: [Semantic conventions for HTTP spans](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.11.0/specification/trace/semantic_conventions/http.md#semantic-conventions-for-http-spans)
+    public var http: HTTPAttributes {
+        get {
+            .init(attributes: self)
+        }
+        set {
+            self = newValue.attributes
+        }
+    }
+}
+```
+
+```swift
+/// Semantic conventions for HTTP spans.
+///
+/// OpenTelemetry Spec: [Semantic conventions for HTTP spans](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.11.0/specification/trace/semantic_conventions/http.md#semantic-conventions-for-http-spans)
+@dynamicMemberLookup
+public struct HTTPAttributes: SpanAttributeNamespace {
+    public var attributes: SpanAttributes
+
+    public init(attributes: SpanAttributes) {
+        self.attributes = attributes
+    }
+
+    public struct NestedSpanAttributes: NestedSpanAttributesProtocol {
+        public init() {}
+
+        /// HTTP request method. E.g. "GET".
+        public var method: Key<String> { "http.method" }
+    }
+}
+```
+
+### Span Events
```

Comment:
Maybe move this section before `Span Attributes` section?

Reply:
Kind of through events are less "common" than attributes, and we do events + logging together, I'll keep as is.